accessibility_api/accessibility_lib/utils/WinUtil.py

```diff
@@ -0,0 +1,217 @@
+                    TARGET.childId = childid
+                    return
+
+        # Traverse through accessible objects of node
+        for child in acc_children:
+            if child not in visited:
+                visited.add(node)
+                WinUtil._traverse(child, visited, search_criteria,
+                                  simple_elements)
+
+    @staticmethod
+    def get_indentifiers(params):
+        """
+        Get accessibility indentifiers
+        """
+
+        identifiers = {
+            'Name': params.get('name'),
+            'Role': params.get('role')
+        }
+        filtered_identifiers = {k: v for k, v in identifiers.items() if v}
+
+        return filtered_identifiers
+
+    @staticmethod
+    def match_criteria(node, params, child_id=CHILDID_SELF):
+        """
+        Find matching accessible with given criteria
+        """
+
+        search_criteria = WinUtil.get_indentifiers(params)
+        for criteria in search_criteria:
+            prefix = 'acc'
+            prop_value = getattr(node, prefix + criteria)(child_id)
+            search_value = search_criteria[criteria]
+
+            # If value is a number convert from unicode to int
+            if isinstance(prop_value, int):
+                search_value = int(search_value)
+
+            if prop_value != search_value:
+                return False
+
+        return True
+
+    @staticmethod
+    def _get_test_window(app_window, simple_elements):
+        """
+        Get test window handle
+        """
+
+        # Get the window for browser
+        test_class = c_char_p('MozillaWindowClass')
+        current_hwnd = windll.user32.FindWindowA(test_class, None)
+        name = (
+            WinUtil._accessible_object_from_window(current_hwnd,
+                                                   simple_elements)
+            .accName(CHILDID_SELF)
+        )
+
+        # Iterate through windows
+        while name is None or app_window not in name:
+            current_hwnd = (
+                windll.user32
+                .FindWindowExA(None, current_hwnd, test_class, None)
+            )
+            name = (
+                WinUtil._accessible_object_from_window(current_hwnd,
+                                                       simple_elements)
+                .accName(CHILDID_SELF)
+            )
+        return current_hwnd
+
+    @staticmethod
+    def get_root_accessible(app_window, simple_elements):
+        """
+        Set root accessible object to test window
+        """
+
+        test_window = WinUtil._get_test_window(app_window, simple_elements)
+        print_test_window(test_window)
+        root = WinUtil._accessible_object_from_window(test_window,
+                                                      simple_elements)
+        return root
+
+    @staticmethod
+    def get_target_accessible(params, simple_elements):
+        """
+        Retrieve the accessible object for the given ID
+        """
+
+        global TARGET
```

Comment:
In fact, lets not add a bunch of arguments, why not pass a structure like this to all winutils from IAccessible etc:
{
  target: None
  visited: set(),
  simple_elements: dict()
} that you will stick on the self of IAccessible similar to how you just do it with simple_elements prop.